olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/VersionedLayerClient.h

```diff
@@ -382,6 +382,45 @@ class DATASERVICE_READ_API VersionedLayerClient final {
    */
   bool IsCached(const geo::TileKey& tile) const;
 
+  /**
+   * @brief Protects tile keys from eviction.
+   *
+   * @note You can only protect tiles which data handles are present in the
+   * cache at the time of the call.
+   *
+   * @note Please do not call `Protect` while the `Release` call for the same
+   * catalog and layer is in progress.
+   *
+   * @note Protecting tile keys mean that its data and corresponding quad tree
+   * keys will be added to protected list and stored in cache. This keys will be
+   * removed from LRU, so they could not be evicted. Also they are not expire.
+   * Quad tree stays protected if at least one tile key is protected.
+   *
+   * @param tiles The list of tile keys to be protected.
+   *
+   * @return True if some keys were successfully added to the protected list;
+   * false otherwise.
+   */
+  bool Protect(const TileKeys& tiles);
+
+  /**
+   * @brief Removes a list of tiles from protection.
+   *
+   * @note Please make sure that `Protect` will not be called for the same
+   * catalog and layer while the `Release` call is in progress.
+   *
+   * @note Releasing tile keys removes data and quad tree keys from protected
+   * list. Keys will be added to LRU, so they could be evicted. Expiration value
+   * is resored and keys could expire. Quad tree could be removed from protected
+   * list, if all tile keys are not longer protected.
```

Comment:
Please move to line 409 (before the note).

```suggestion
   * Releasing tile keys removes data and quadtree keys from the protected
   * list. The keys are added to the LRU cache, so they could be evicted. Expiration value
   * is restored, and keys can expire. The quadtree can be removed from the protected
   * list if all tile keys are no longer protected.
```